flight/Modules/Actuator/actuator.c

```diff
@@ -297,126 +277,36 @@ static void actuatorTask(void* parameters)
-				curve2 = CollectiveCurve(desired.Throttle, mixerSettings.ThrottleCurve2, MIXERSETTINGS_THROTTLECURVE2_NUMELEM);
-				break;
-			case MIXERSETTINGS_CURVE2SOURCE_ROLL:
-				curve2 = CollectiveCurve(desired.Roll, mixerSettings.ThrottleCurve2, MIXERSETTINGS_THROTTLECURVE2_NUMELEM);
-				break;
-			case MIXERSETTINGS_CURVE2SOURCE_PITCH:
-				curve2 = CollectiveCurve(desired.Pitch, mixerSettings.ThrottleCurve2, MIXERSETTINGS_THROTTLECURVE2_NUMELEM);
-				break;
-			case MIXERSETTINGS_CURVE2SOURCE_YAW:
-				curve2 = CollectiveCurve(desired.Yaw, mixerSettings.ThrottleCurve2, MIXERSETTINGS_THROTTLECURVE2_NUMELEM);
-				break;
-			case MIXERSETTINGS_CURVE2SOURCE_COLLECTIVE:
-				ManualControlCommandCollectiveGet(&curve2);
-				curve2 = CollectiveCurve(curve2, mixerSettings.ThrottleCurve2, MIXERSETTINGS_THROTTLECURVE2_NUMELEM);
-				break;
-			case MIXERSETTINGS_CURVE2SOURCE_ACCESSORY0:
-			case MIXERSETTINGS_CURVE2SOURCE_ACCESSORY1:
-			case MIXERSETTINGS_CURVE2SOURCE_ACCESSORY2:
-			case MIXERSETTINGS_CURVE2SOURCE_ACCESSORY3:
-			case MIXERSETTINGS_CURVE2SOURCE_ACCESSORY4:
-			case MIXERSETTINGS_CURVE2SOURCE_ACCESSORY5:
-				if(AccessoryDesiredInstGet(mixerSettings.Curve2Source - MIXERSETTINGS_CURVE2SOURCE_ACCESSORY0,&accessory) == 0)
-					curve2 = CollectiveCurve(accessory.AccessoryVal, mixerSettings.ThrottleCurve2, MIXERSETTINGS_THROTTLECURVE2_NUMELEM);
-				else
-					curve2 = 0;
-				break;
-		}
+		float curve2 = CollectiveCurve(
+			GetCurve2Source(&desired, mixerSettings.Curve2Source),
+			mixerSettings.ThrottleCurve2,
+			MIXERSETTINGS_THROTTLECURVE2_NUMELEM);
+
 
 		float * status = (float *)&mixerStatus; //access status objects as an array of floats
 
 		for(int ct=0; ct < MAX_MIX_ACTUATORS; ct++)
 		{
-			if(mixers[ct].type == MIXERSETTINGS_MIXER1TYPE_DISABLED) {
-				// Set to minimum if disabled.  This is not the same as saying PWM pulse = 0 us
-				status[ct] = -1;
-				command.Channel[ct] = 0.0f;
-				continue;
-			}
-
-			if((mixers[ct].type == MIXERSETTINGS_MIXER1TYPE_MOTOR) || (mixers[ct].type == MIXERSETTINGS_MIXER1TYPE_SERVO)) {
-				status[ct] = ProcessMixer(ct, curve1, curve2, &mixerSettings, &desired, dT);
-
-				// If a motor type, additional work to be done
-				if(mixers[ct].type == MIXERSETTINGS_MIXER1TYPE_MOTOR) {
-					if (status[ct] > 0) {
-						// Apply curve fitting, mapping the input to the propeller output.
-						status[ct] = actuatorSettings.MotorInputOutputCurveFit[ACTUATORSETTINGS_MOTORINPUTOUTPUTCURVEFIT_A] *
-						      powf(status[ct], actuatorSettings.MotorInputOutputCurveFit[ACTUATORSETTINGS_MOTORINPUTOUTPUTCURVEFIT_B]);
-					} else {
-						 // Idle throttle
-						status[ct] = 0.0f;
-					}
-				}
-			} else {
-				status[ct] = -1;
-			}
+			status[ct] = MixChannel(ct, &desired, curve1, curve2);
 
 			// Motors have additional protection for when to be on
-			if(mixers[ct].type == MIXERSETTINGS_MIXER1TYPE_MOTOR) {
+			if(GetMixerType(ct) == MIXERSETTINGS_MIXER1TYPE_MOTOR) {
```

Comment:
This is evaluated twice in the ```while``` loop, here and at https://github.com/TauLabs/TauLabs/pull/1987/files#diff-6756853a8b52a94d64d1f12e4377cb31R260. Do you see a way to reduce that to just once?